hopsworks-common/src/main/java/io/hops/hopsworks/common/featurestore/featuregroup/ondemand/OnDemandFeaturegroupController.java

```diff
@@ -90,94 +81,21 @@ public OnDemandFeaturegroup createOnDemandFeaturegroup(OnDemandFeaturegroupDTO o
   @TransactionAttribute(TransactionAttributeType.NEVER)
   public void updateOnDemandFeaturegroupMetadata(OnDemandFeaturegroup onDemandFeaturegroup,
     OnDemandFeaturegroupDTO onDemandFeaturegroupDTO) throws FeaturestoreException {
+    // Verify User Input specific for on demand feature groups
+    FeaturestoreJdbcConnector featurestoreJdbcConnector =
+      verifyOnDemandFeaturegroupJdbcConnector(onDemandFeaturegroupDTO.getJdbcConnectorId());
+    verifyOnDemandFeaturegroupSqlQuery(onDemandFeaturegroupDTO.getQuery());
     
-    if(!Strings.isNullOrEmpty(onDemandFeaturegroupDTO.getName())){
-      verifyOnDemandFeaturegroupName(onDemandFeaturegroupDTO.getName());
-      onDemandFeaturegroup.setName(onDemandFeaturegroupDTO.getName());
-    }
-    if(!Strings.isNullOrEmpty(onDemandFeaturegroupDTO.getDescription())){
-      verifyOnDemandFeaturegroupDescription(onDemandFeaturegroupDTO.getDescription());
-      onDemandFeaturegroup.setDescription(onDemandFeaturegroupDTO.getDescription());
-    }
-    if(onDemandFeaturegroupDTO.getJdbcConnectorId() != null) {
-      FeaturestoreJdbcConnector featurestoreJdbcConnector =
-        verifyOnDemandFeaturegroupJdbcConnector(onDemandFeaturegroupDTO.getJdbcConnectorId());
-      onDemandFeaturegroup.setFeaturestoreJdbcConnector(featurestoreJdbcConnector);
-    }
-    if(!Strings.isNullOrEmpty(onDemandFeaturegroupDTO.getQuery())){
-      verifyOnDemandFeaturegroupSqlQuery(onDemandFeaturegroupDTO.getQuery());
-    }
-    //Update metadata in on_demand_feature_group table
-    onDemandFeaturegroupFacade.updateMetadata(onDemandFeaturegroup);
+    // Update metadata in entity
+    onDemandFeaturegroup.setName(onDemandFeaturegroupDTO.getName());
+    onDemandFeaturegroup.setDescription(onDemandFeaturegroupDTO.getDescription());
+    onDemandFeaturegroup.setFeaturestoreJdbcConnector(featurestoreJdbcConnector);
+    onDemandFeaturegroup.setQuery(onDemandFeaturegroupDTO.getQuery());
```

Comment:
I think that the query update is dangerous. Changing the query can lead to a completely different feature group, potentially breaking downstream jobs that build training datasets.

Reply:
This is how it was, I just put all validation at the beginning. I didn't want to change too much about the behaviour of the updateMetadata method in this PR. We can fix it in another PR. But as we discussed this method shouldn't be there in the first place.